lib/mapper_tools.py

```diff
@@ -85,53 +86,56 @@ def loadMapperData(mode = 'train', data_dir_train = DATA_DIR_TRAIN, data_dir_tes
     return data_train, data_header_train, data_test, data_header_test
 
 def getLatentRep(n_components, data, remake=True):
-    label = "PCA%d" % (n_components)
-    if not remake and os.path.exists("%s_latent" % (label)):
-        frin = open(TEMP_DATA+"%s_latent" % (label), "rb")
-        rep = pickle.load( frin )
+    label = "PCA%d" % n_components
+    if not remake and os.path.exists("%s_latent" % label):
+        frin = open(TEMP_DATA+"%s_latent" % label, "rb")
+        rep = pickle.load(frin)
         return rep
 
+    if log.level == logging.DEBUG:
+        print("------> fitting PCA to data of shape {}...".format(data.shape))
     pca = MyPCA(n_components=n_components)
     pca.fit(data)
 
+
     rep = LatentRep([pca], label)
-    fr = open(TEMP_DATA+"%s_latent" % (label), "wb")
+    fr = open(TEMP_DATA+"%s_latent" % label, "wb")
     pickle.dump(rep, fr)
     fr.close()
 
     return rep
 
 def runMapper(n_components, data, rep, remake=True, verbose=0):
-    label = "PCA%d" % (n_components)
-    if not remake and os.path.exists("%s_firstsimplegap_graphs" % (label)):
-        fgin = open(TEMP_DATA+"%s_firstsimplegap_graphs" % (label), "rb")
+    label = "PCA%d" % n_components
+    if not remake and os.path.exists("%s_firstsimplegap_graphs" % label):
+        fgin = open(TEMP_DATA+"%s_firstsimplegap_graphs" % label, "rb")
         graphs = pickle.load(fgin)
 
-        fpin = open(TEMP_DATA+"%s_mapper_pipes" % (label), "rb")
+        fpin = open(TEMP_DATA+"%s_mapper_pipes" % label, "rb")
         mapper_pipes = pickle.load(fpin)
 
         return graphs, mapper_pipes
 
     clusterer = FirstSimpleGap()
     mapper_pipes = []
-    pca=rep.projectors[0]
+    pca = rep.projectors[0]
 
     if log.level == logging.DEBUG:
         print("------> creating projection components...")
     for k in range(n_components):
         if log.level == logging.DEBUG:
-            print("---------> on component {}/{}...".format(k+1,n_components))
-        proj = Projection(columns = k)
+            print("---------> on component {}/{}...".format(k+1, n_components))
```

Comment:
Use logging here, e.g. log.info("---------> on component {}/{}...".format(k+1, n_components)), or whatever is the appropriate level

Minor point: using an fstring might make this more readable: f"---------> on component {k+1}/{n_components}..."